dropwizard-auth/src/main/java/io/dropwizard/auth/CachingAuthenticator.java

```diff
@@ -60,7 +60,12 @@ public CachingAuthenticator(final MetricRegistry metricRegistry,
             @Override
             public Optional<P> load(C key) throws Exception {
                 cacheMisses.mark();
-                return underlying.authenticate(key);
+                final Optional<P> result = underlying.authenticate(key);
+                // only cache present values
+                if (!result.isPresent()) {
+                    throw new AuthenticationException("Failed to load security context into cache");
```

Comment:
Won't this exception be thrown (and propagated!) every time invalid credentials are provided?